modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java

```diff
@@ -280,12 +283,38 @@ private static void _validateNewValue(
 			Object existingValue, String objectFieldName, Object value)
 		throws PortalException {
 
-		if (!((Validator.isNull(existingValue) && Validator.isNull(value)) ||
-			  Objects.equals(existingValue, value))) {
+		if (Validator.isNull(existingValue) && Validator.isNull(value)) {
+			return;
+		}
+
+		if ((value instanceof Boolean) &&
+			Objects.equals(
+				GetterUtil.getBoolean(value),
+				GetterUtil.getBoolean(existingValue))) {
```

Comment:
I don't know if it's a possible scenario but if `existingValue` was not a instance of boolean, GetterUtil.getBoolean will return false value.